sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -1016,20 +1031,12 @@ private async Task ReconcileConcurrencyAsync()
             // decreasing concurrency
             else if (diff < 0)
             {
-                var activeTasks = _tasks.Where(t => !t.Task.IsCompleted).ToList();
+                var activeTasks = TaskTuples.Where(t => !t.Task.IsCompleted).ToList();
                 int excessTasks = activeTasks.Count - _maxConcurrentCalls;
 
                 // cancel excess tasks
                 for (int i = 0; i < excessTasks; i++)
                 {
-                    if (IsSessionProcessor)
-                    {
-                        // Session managers have CTS that are managed internally when a new
```

Comment:
This isn't necessary and actually is wrong since you can have more than 1 task receiving from the same session.